x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -133,8 +137,8 @@ public final PhysicalOperation groupingPhysicalOperation(
                             // is in the output form
                             // if the group points to an alias declared in the aggregate, use the alias child as source
                             else if (aggregatorMode.isOutputPartial()) {
-                                if (groupAttribute.semanticEquals(a.toAttribute())) {
-                                    groupAttribute = attr;
+                                if (sourceGroupAttribute.semanticEquals(a.toAttribute())) {
+                                    sourceGroupAttribute = attr;
                                     break;
```

Comment:
The `groupAttribute` was mutated, but we use the `sourceGroupAttribute` to determine the right channel to pass to the `groupSpecs`. We need to mutate the `sourceGroupAttribute` instead.